datastore/datastore.go

```diff
@@ -0,0 +1,217 @@
+package datastore
+
+import (
+	"bytes"
+	"encoding/gob"
+	"errors"
+	"reflect"
+	"strings"
+
+	"github.com/docker/libnetwork/config"
+	"github.com/docker/swarm/pkg/store"
+)
+
+//DataStore exported
+type DataStore interface {
+	// PutObject adds a new Record based on an object into the datastore
+	PutObject(kvObject KV) error
+	// PutObjectAtomic provides an atomic add and update operation for a Record
+	PutObjectAtomic(kvObject KV) error
+	// KVStore returns access to the KV Store
+	KVStore() store.Store
+}
+
+type datastore struct {
+	store store.Store
+}
+
+//KV Key Value interface used by objects to be part of the DataStore
+type KV interface {
+	Key() []string
```

Comment:
Please add comments to interface methods on what it does and what the side effects are?